### PR TITLE
Fetch time entries created "today" on the initial pull sync

### LINKED ISSUE #2451: Time entries from other workspace aren't displayed when the user log in
It was reported that after log in, only TEs from the default WS are displayed, though we should display TEs across all the WSs.
Atm it requires manual sync to see TE from other WSs.

Can someone reproduce it? In my case, the app downloads TEs from all the workspaces right after the log in.

### PR DESCRIPTION
**Pull request by daividssilverio**
Closes #2451

Please check #2451 for more context on this change. 
Basically, the backend doesn't fetch the time entries from the end_date param we are sending in the request. This simple change fixes that.

~🦑 whenever~

### CODE REVIEW

File: Toggl.Foundation/Sync/States/Pull/FetchAllSinceState.cs [hunk 1]
```diff
@@ -16,6 +16,7 @@ internal sealed class FetchAllSinceState : ISyncState
         private readonly ITimeService timeService;
         private const int sinceDateLimitMonths = 2;
         private const int fetchTimeEntriesForMonths = 2;
+        private const int timeEntriesEndDateInclusideExtraDaysCount = 2;
```
Comment (kornelijepetak) [inline Toggl.Foundation/Sync/States/Pull/FetchAllSinceState.cs]: `InclusiDe` typo

File: Toggl.Ultrawave.Tests.Integration/TimeEntriesApiTests.cs [hunk 2]
```diff
@@ -124,6 +124,44 @@ protected override IObservable<List<ITimeEntry>> CallEndpointWith(ITogglApi togg
...
+                list.Should().HaveCount(1);
+                var item = list[0];
+                item.Start.ToUnixTimeSeconds().Should().BeGreaterOrEqualTo(timeEntryFiveMinuteBeforeTheBeginningOfTheEndDate.Start.AddMinutes(-1).ToUnixTimeSeconds());
+                item.Start.ToUnixTimeSeconds().Should().BeLessThan(timeEntryFiveMinuteBeforeTheBeginningOfTheEndDate.Start.AddMinutes(1).ToUnixTimeSeconds());
+            }
+
+            [Fact]
+            public async Task ReturnsTimeEntriesOnlyAfterBeginningOfTheStartDate()
+            {
+
```
Comment (kornelijepetak) [inline Toggl.Ultrawave.Tests.Integration/TimeEntriesApiTests.cs]: :fire: `\n`

File: Toggl.Ultrawave.Tests.Integration/TimeEntriesApiTests.cs [hunk 3]
```diff
@@ -124,6 +124,44 @@ protected override IObservable<List<ITimeEntry>> CallEndpointWith(ITogglApi togg
                 item.Start.ToUnixTimeSeconds().Should().BeGreaterOrEqualTo(start.ToUnixTimeSeconds());
                 item.Start.ToUnixTimeSeconds().Should().BeLessThan(end.ToUnixTimeSeconds());
             }
+
+            [Fact]
+            public async Task ReturnsTimeEntriesOnlyUpToBeginningOfTheEndDate()
+            {
+
+                var theBeginningOfEndDate = end.AddHours(-end.Hour + end.Offset.Hours).AddMinutes(-end.Minute).AddSeconds(-end.Second + 1);
```
Comment (amulware) [inline Toggl.Ultrawave.Tests.Integration/TimeEntriesApiTests.cs]: I'm pretty sure there must be an easier way to do this.

File: Toggl.Ultrawave.Tests.Integration/TimeEntriesApiTests.cs [hunk 4]
```diff
@@ -124,6 +124,44 @@ protected override IObservable<List<ITimeEntry>> CallEndpointWith(ITogglApi togg
...
+                var timeEntryFiveMinuteBeforeTheBeginningOfTheEndDate = createTimeEntry(user, theBeginningOfEndDate.AddMinutes(-5));
+                await api.TimeEntries.Create(timeEntryRightAtTheBeginningOfTheEndDate);
+                await api.TimeEntries.Create(timeEntryFiveMinuteBeforeTheBeginningOfTheEndDate);
+
+                var list = await api.TimeEntries.GetAll(start, end);
+
+                list.Should().HaveCount(1);
+                var item = list[0];
+                item.Start.ToUnixTimeSeconds().Should().BeGreaterOrEqualTo(timeEntryFiveMinuteBeforeTheBeginningOfTheEndDate.Start.AddMinutes(-1).ToUnixTimeSeconds());
+                item.Start.ToUnixTimeSeconds().Should().BeLessThan(timeEntryFiveMinuteBeforeTheBeginningOfTheEndDate.Start.AddMinutes(1).ToUnixTimeSeconds());
```
Comment (amulware) [inline Toggl.Ultrawave.Tests.Integration/TimeEntriesApiTests.cs]: Isn't an exact equality comparison possible here? Backend should not touch start times we give it.

File: Toggl.Ultrawave.Tests.Integration/TimeEntriesApiTests.cs [hunk 5]
```diff
@@ -124,6 +124,44 @@ protected override IObservable<List<ITimeEntry>> CallEndpointWith(ITogglApi togg
...
+            }
+
+            [Fact]
+            public async Task ReturnsTimeEntriesOnlyAfterBeginningOfTheStartDate()
```
Comment (amulware) [inline Toggl.Ultrawave.Tests.Integration/TimeEntriesApiTests.cs]: Same comments for this test.

File: Toggl.Ultrawave.Tests.Integration/TimeEntriesApiTests.cs [hunk 6]
```diff
@@ -124,6 +124,44 @@ protected override IObservable<List<ITimeEntry>> CallEndpointWith(ITogglApi togg
                 item.Start.ToUnixTimeSeconds().Should().BeGreaterOrEqualTo(start.ToUnixTimeSeconds());
                 item.Start.ToUnixTimeSeconds().Should().BeLessThan(end.ToUnixTimeSeconds());
             }
+
+            [Fact]
+            public async Task ReturnsTimeEntriesOnlyUpToBeginningOfTheEndDate()
+            {
+
```
Comment (simonrozsival) [inline Toggl.Ultrawave.Tests.Integration/TimeEntriesApiTests.cs]: 👀

File: Toggl.Ultrawave.Tests.Integration/TimeEntriesApiTests.cs [hunk 7]
```diff
@@ -124,6 +124,44 @@ protected override IObservable<List<ITimeEntry>> CallEndpointWith(ITogglApi togg
...
+                var theBeginningOfEndDate = end.AddHours(-end.Hour + end.Offset.Hours).AddMinutes(-end.Minute).AddSeconds(-end.Second + 1);
+                var (api, user) = await SetupTestUser();
+                var timeEntryRightAtTheBeginningOfTheEndDate = createTimeEntry(user, theBeginningOfEndDate);
```
Comment (simonrozsival) [inline Toggl.Ultrawave.Tests.Integration/TimeEntriesApiTests.cs]: remove the `TheBeginningOf` from the name, the current names are confusing (also in the other cases of variables containing `TheBeginningOfEnd` or `TheBeginningOfStart`).

**Comment by amulware** [comment]
Btw, do the integration tests pass on this? Last night's tests on `develop` failed for this endpoint I believe.

File: Toggl.Ultrawave.Tests.Integration/TimeEntriesApiTests.cs (hunk 3 above)
Comment (willsb) [reply Toggl.Ultrawave.Tests.Integration/TimeEntriesApiTests.cs]: `var theBeginningOfEndDate = end.Date;`

File: Toggl.Ultrawave.Tests.Integration/TimeEntriesApiTests.cs (hunk 7 above)
Comment (daividssilverio) [reply Toggl.Ultrawave.Tests.Integration/TimeEntriesApiTests.cs]: 🤔 I wanted to make sure those meant that date with hours:minutes:seconds set to 00:00:01.

**Comment by daividssilverio** [comment]
https://www.bitrise.io/build/949e860bca668133

**Comment by amulware** [comment]
Oh, I did. But didn't re-review. Duh.

**Pull request merged**